compiler/rustc_error_codes/src/error_codes/E0267.md
===================================================

Last edited: 2023-03-30 20:35:59

Contents:

.. code-block:: md

    A loop keyword (`break` or `continue`) was used inside a closure but outside of
any loop.

Erroneous code example:

```compile_fail,E0267
let w = || { break; }; // error: `break` inside of a closure
```

`break` and `continue` keywords can be used as normal inside closures as long as
they are also contained within a loop. To halt the execution of a closure you
should instead use a return statement. Example:

```
let w = || {
    for _ in 0..10 {
        break;
    }
};

w();
```


